### PR TITLE
Adds Cookbook for Manhattan distance.

### PR DESCRIPTION
**Pull request by arianepaola**
This adds a cookbook for Manhattan distance.

### CODE REVIEW

**Comment by arianepaola** [comment]
@vigsterkr @karlnapf @lgoetz

Please review and comment. I would like to improve and merge this cookbook.

**Comment by arianepaola** [comment]
@vigsterkr @karlnapf @lgoetz

Please review and comment. I would like to improve and merge this cookbook.

File: examples/meta/src/distance/manhattan_distance.sg [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+RealFeatures features_train(f_feats_train)
+RealFeatures features_test(f_feats_test)
+#![create_features]
+
+#![create_distance]
+ManhattanMetric distance(features_train, features_train)
+#![create_distance]
+
+#![train_and_init]
+DoubleMatrix dm_train = distance.get_distance_matrix()
```
Comment (karlnapf) [inline examples/meta/src/distance/manhattan_distance.sg]: use `RealMatrix`

File: examples/meta/src/distance/manhattan_distance.sg [hunk 2]
```diff
@@ -0,0 +1,18 @@
+CSVFile f_feats_train("../../data/fm_train_real.dat")
```
Comment (karlnapf) [inline examples/meta/src/distance/manhattan_distance.sg]: not sure "train" and "test" is the best name here. You never train or test with a model.
We want to illustrate two things: Evaluating a symmetric distance, and one between different data.
Therefore, just call them features_a, features_b maybe

File: examples/meta/src/distance/manhattan_distance.sg [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+#![create_features]
+
+#![create_distance]
+ManhattanMetric distance(features_train, features_train)
+#![create_distance]
+
+#![train_and_init]
+DoubleMatrix dm_train = distance.get_distance_matrix()
+distance.init(features_train, features_test)
+DoubleMatrix dm_test = distance.get_distance_matrix()
```
Comment (karlnapf) [inline examples/meta/src/distance/manhattan_distance.sg]: Can you also show the interface to call the distance() method itself (that returns a float)

**Comment by karlnapf** [comment]
This also needs integration testing data. Let me know if you need help with that. It is also described in the cookbook readme

**Comment by arianepaola** [comment]
@karlnapf Thanks for the comment. I will update this PR.

**Comment by karlnapf** [comment]
Looking forward to it!

File: examples/meta/src/distance/manhattan_distance.sg (hunk 3 above)
Comment (arianepaola) [inline examples/meta/src/distance/manhattan_distance.sg]: @karlnapf This one?
distance(features_a, features_b)

**Comment by karlnapf** [comment]
This still needs integration testing data, ping me if you have questions

File: examples/meta/src/distance/manhattan_distance.sg [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+RealFeatures features_a(f_feats_a)
+RealFeatures features_b(f_feats_b)
+#![create_features]
+
+#![create_distance]
+ManhattanMetric distance(features_a, features_b)
+#![create_distance]
+
+#![train_and_init]
+RealMatrix dm_train = distance.get_distance_matrix()
```
Comment (karlnapf) [inline examples/meta/src/distance/manhattan_distance.sg]: call this one `distance_matrix_ab`

File: examples/meta/src/distance/manhattan_distance.sg [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+distance.init(features_a, features_b)
```
Comment (karlnapf) [inline examples/meta/src/distance/manhattan_distance.sg]: this is the same as above, so both of your distance matrices will be the same.
Actually, why don't we just remove everything after this line, and only compute the distance_ab.
It is pretty obvious how to do other combinations.....

**Comment by arianepaola** [comment]
@karlnapf Thanks for the updated comments.

**Comment by karlnapf** [comment]
Any updates here

**Comment by arianepaola** [comment]
@karlnapf will be updating this PR.

**Pull request closed without merging**